src/ansys/materials/manager/util/matml/utils.py

```diff
@@ -0,0 +1,183 @@
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import re
+from typing import Dict, Sequence
+import xml.etree.ElementTree as ET
+
+
+def parse_unit_string(unit_str: str) -> list[tuple[str, int]]:
+    """Parse a unit string into a list of tuples."""
+    if unit_str.strip().lower() == "unitless":
+        return [("Unitless", 1)]
+
+    pattern = r"([a-zA-Z]+)(?:\^?(-?\d+))?"
+    units = unit_str.split(" ")
+    result = []
+
+    for unit in units:
+        match = re.fullmatch(pattern, unit.strip())
+        if match:
+            name = match.group(1)
+            power = int(match.group(2)) if match.group(2) else 1
+            result.append((name, power))
+        else:
+            raise ValueError(f"Invalid unit format: {unit}")
+
+    return result
+
+
+def xml_to_unit(param: ET.Element) -> tuple[str, dict[str, str]]:
+    """
+    Convert XML element to a unit string.
+
+    Parameters
+    ----------
+    param : ET.Element
+        The XML element containing the unit information.
+
+    Returns
+    -------
+    tuple[str, dict[str, str]]
+        A tuple containing the unit ID and a dictionary with unit details.
+    """
+    id = param.attrib.get("id", "")
+    entry = {"Name": param.findtext("Name", ""), "UnitsName": "", "Units": ""}
+    units = param.find("Units")
+    if units is not None:
+        entry["UnitsName"] = units.attrib.get("name", "")
+        unit_parts = []
+        for unit in units.findall("Unit"):
+            name = unit.findtext("Name", "")
+            power = unit.attrib.get("power")
+            if power:
+                unit_parts.append(f"{name}^{power}")
+            else:
+                unit_parts.append(name)
+        entry["Units"] = " ".join(unit_parts)
+    elif param.find("Unitless") is not None:
+        entry["Units"] = "Unitless"
+    return id, entry
+
+
+def units_to_xml(unit: str) -> ET.Element:
```

Comment:
```suggestion
def unit_to_xml(unit: str) -> ET.Element:
```
given that the argument is a single unit?